save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/service/DockerService.kt

```diff
@@ -39,6 +42,8 @@ class DockerService(
     private val agentRunner: AgentRunner,
     private val agentService: AgentService,
 ) {
+    private val areAgentsHaveStarted: ConcurrentMap<Long, AtomicBoolean> = ConcurrentHashMap()
```

Comment:
`AtomicReference<Map<Long, AtomicBoolean>>`?
@nulls what do you suggest?

Reply:
`ConcurrentHashMap` allows concurrent access to different keys, while `AtomicReference<Map<*, *>>` will restrict access to a single thread only, I think

Reply:
agree about `ConcurrentHashMap`